File: src/main/webapp/resources/service/teacher/SubjectService.js
App.factory('SubjectService', ['$http', '$q', function($http, $q){
	 
	var REST_SERVICE_URI = 'http://localhost:8080/EvaluationSource/api/';
 
    var factory = {
    	fetchAllTargetBySubjectIdChapterId1: fetchAllTargetBySubjectIdChapterId1,
    	fetchAllTargetBySubjectIdChapterId: fetchAllTargetBySubjectIdChapterId,
    	fetchExamsResult: fetchExamsResult,
    	fetchAllChapterBySubjectId: fetchAllChapterBySubjectId,
    	fetchAllTargetBySubjectId: fetchAllTargetBySubjectId,
    	fetchAllExamsBySubjectId: fetchAllExamsBySubjectId,
    	fetchAllClassBySubjectId: fetchAllClassBySubjectId,
    	fetchAllStudentBySubjectId: fetchAllStudentBySubjectId,
    	fetchSubjectBySubjectId: fetchSubjectBySubjectId,
    	isSuperTeacherOfSubject: isSuperTeacherOfSubject,
    	assignStudentToSubject: assignStudentToSubject,
    	createChapter: createChapter,
    	createTarget: createTarget,
    	assign: assign,
    	fileStudentToSubject: fileStudentToSubject,
    	approveExams: approveExams,
    	removeStudentFromSubject: removeStudentFromSubject,
    	removeTargetFromSubject: removeTargetFromSubject,
    	removeChapterFromSubject: removeChapterFromSubject,
    	closeClass: closeClass,
    };
 
    return factory;
    
    function fetchExamsResult(subjectId, examsId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'fetchExamsResult/' + subjectId + '/' + examsId).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function closeClass(subjectId, classId) {
    	var deferred = $q.defer();
    	const data = {
    			subjectid: subjectId,
    			id: classId,
    	};
    	$http.post(REST_SERVICE_URI + 'closeClass', data).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function removeTargetFromSubject(subjectId, targetId) {
    	var deferred = $q.defer();
    	const data = {
    			subjectid: subjectId,
    			id: targetId,
    	};
    	console.log(" TARGET");
    	console.log(data);
    	$http.post(REST_SERVICE_URI + 'removeTargetFromSubject', data).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function removeChapterFromSubject(subjectId, chapterId) {
    	var deferred = $q.defer();
    	const data = {
    			subjectid: subjectId,
    			id: chapterId,
    	};
    	$http.post(REST_SERVICE_URI + 'removeChapterFromSubject', data).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function removeStudentFromSubject(studentId, subjectId) {
    	var deferred = $q.defer();
    	const data = {
    			studentid: studentId,
    			subjectid: subjectId,
    	};
    	$http.post(REST_SERVICE_URI + 'removeStudentFromSubject', data).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function approveExams(examsId) {
    	var deferred = $q.defer();
    	$http.post(REST_SERVICE_URI + 'approveExams', examsId).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function assign(examsId, classId) {
    	var deferred = $q.defer();
    	var data = {
    		classid: classId,
    		examsid: examsId,
    	};
    	$http.post(REST_SERVICE_URI + 'assign', data).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    /************ Insert Function: add student to subject **************/
    function assignStudentToSubject(subjectId, studentEmail) {
    	var deferred = $q.defer();
    	var data = {
    		email: studentEmail,
    	};
    	$http.post(REST_SERVICE_URI + 'checkStudent', data).then(function (response) {
                deferred.resolve(response.data);
                console.log(response.data);
                if (response.data.id) {
                	data = {
                		studentid: response.data.id,
                		subjectid: subjectId,
                	};
                	$http.post(REST_SERVICE_URI + 'assignStudentToSubject', data).then(function (response) {
		                    deferred.resolve(response.data);
		            	}, function(errResponse){
		                    console.error(errResponse);
		                    deferred.reject(errResponse);
		            	}
		            );
                }
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    };
    
    /************ Insert Function: create chapter of subject **************/
    function createChapter(subjectId, chapterName) {
    	var deferred = $q.defer();
    	var chapter = {
    		name: chapterName,
    		subjectid: subjectId,
    	};
    	console.log(chapter);
    	$http.post(REST_SERVICE_URI + 'createChapter', chapter).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    };
    
    /************ Insert Function: create target of subject **************/
    function createTarget(subjectId, chapterId, targetName) {
    	var deferred = $q.defer();
    	const data = {
    		content: targetName,
    		subjectid: subjectId,
    		chapterid: chapterId,
    	};
    	$http.post(REST_SERVICE_URI + 'createTarget', data).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    };
    
    /************ Check Function: current teacher is super teacher of subject **************/
    function isSuperTeacherOfSubject(subjectId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'isSuperTeacherOfSubject/' + subjectId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    
    /************ Fetch function **************/
    
    function fetchAllChapterBySubjectId(subjectId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getAllChapterBySubjectId/' + subjectId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fetchAllTargetBySubjectIdChapterId(subjectId, chapterId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getAllTargetBySubjectIdChapterId/' + subjectId + '/' + chapterId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fetchAllTargetBySubjectIdChapterId1(subjectId, chapterId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getAllTargetBySubjectIdChapterId1/' + subjectId + '/' + chapterId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fetchAllTargetBySubjectId(subjectId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getAllTargetBySubjectId/' + subjectId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fetchAllExamsBySubjectId(subjectId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getAllExamsBySubjectId/' + subjectId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fetchAllStudentBySubjectId(subjectId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getAllStudentBySubjectId/' + subjectId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fetchAllClassBySubjectId(subjectId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getAllClassBySubjectId/' + subjectId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fetchSubjectBySubjectId(subjectId) {
    	var deferred = $q.defer();
    	$http.get(REST_SERVICE_URI + 'getSubjectBySubjectId/' + subjectId).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
    
    function fileStudentToSubject(subjectId, csvContent) {
    	var deferred = $q.defer();
    	const data = {
    		subjectId: subjectId,
    		content: csvContent,
    	}
    	$http.post(REST_SERVICE_URI + 'fileStudentToSubject', data).then(function (response) {
                deferred.resolve(response.data);
        	}, function(errResponse){
                console.error(errResponse);
                deferred.reject(errResponse);
        	}
        );
        return deferred.promise;
    }
}]);

File: src/main/java/thanhtuu/springmvc/Service/SubjectService.java
package thanhtuu.springmvc.Service;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.StringReader;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import org.apache.ibatis.session.SqlSession;
import org.apache.ibatis.session.SqlSessionFactory;

import thanhtuu.springmvc.Dao.ChapterMapper;
import thanhtuu.springmvc.Dao.Class_MemberMapper;
import thanhtuu.springmvc.Dao.ClassesMapper;
import thanhtuu.springmvc.Dao.Classes_ExamsMapper;
import thanhtuu.springmvc.Dao.ExamsMapper;
import thanhtuu.springmvc.Dao.QuestionsMapper;
import thanhtuu.springmvc.Dao.Questions_TargetMapper;
import thanhtuu.springmvc.Dao.StudentMapper;
import thanhtuu.springmvc.Dao.Student_ExamMapper;
import thanhtuu.springmvc.Dao.Student_SubjectMapper;
import thanhtuu.springmvc.Dao.SubjectMapper;
import thanhtuu.springmvc.Dao.Super_Teacher_SubjectMapper;
import thanhtuu.springmvc.Dao.TargetMapper;
import thanhtuu.springmvc.Domain.Chapter;
import thanhtuu.springmvc.Domain.Class_Member;
import thanhtuu.springmvc.Domain.Classes;
import thanhtuu.springmvc.Domain.Classes_Exams;
import thanhtuu.springmvc.Domain.Exams;
import thanhtuu.springmvc.Domain.Questions;
import thanhtuu.springmvc.Domain.Questions_TargetKey;
import thanhtuu.springmvc.Domain.Student;
import thanhtuu.springmvc.Domain.Student_Exam;
import thanhtuu.springmvc.Domain.Student_Subject;
import thanhtuu.springmvc.Domain.Subject;
import thanhtuu.springmvc.Domain.Super_Teacher_Subject;
import thanhtuu.springmvc.Domain.Target;
import thanhtuu.springmvc.Temporary.TargetList;

public class SubjectService implements SubjectServiceLocal{
	@Override
	public int createChapter(Chapter chapter) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		ChapterMapper mapper = sqlSession.getMapper(ChapterMapper.class);
		Date now = new Date();
		chapter.setModifiedat(now);
		chapter.setCreatedat(now);
		chapter.setIsactive(true);
		int result = mapper.insertSelective(chapter);
		sqlSession.close();
		
		return result;
	}
	
	@Override
	public int createTarget(Target target) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		System.out.println(target.getContent());
		TargetMapper targetMapper = sqlSession.getMapper(TargetMapper.class);
		target.setIsactive(true);
		int result = targetMapper.insertSelective(target);
		
		sqlSession.close();
		return result;
	}
	
	@Override
	public boolean isSuperTeacherOfSubject(long teacherId, long subjectId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		Super_Teacher_SubjectMapper mapper = sqlSession.getMapper(Super_Teacher_SubjectMapper.class);
		List<Super_Teacher_Subject> result = mapper.isSuperTeacherOfSubject(teacherId, subjectId);
		sqlSession.close();
		
		if (result.size() >= 1) return true;
		else return false;
	}
	
	@Override
	public int assignStudentToSubject(Student_Subject studentSubject) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		Student_SubjectMapper mapper = sqlSession.getMapper(Student_SubjectMapper.class);
		
		int result = 0;
		Date now = new Date();
		studentSubject.setModifiedat(now);
		studentSubject.setIsactive(true);
		
		if (mapper.selectByPrimaryKey(studentSubject) == null) {
			studentSubject.setCreatedat(now);
			result = mapper.insertSelective(studentSubject);
		} else {
			result = mapper.updateByPrimaryKeySelective(studentSubject);
		}
		
		sqlSession.close();
		return result;
	}
	
	/************ Fetch function **************/
	@Override
	public List<Target> getAllTargetBySubjectIdChapterId(long subjectId, long chapterId){
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		TargetMapper mapper = sqlSession.getMapper(TargetMapper.class);
		List<Target> result = mapper.getBySubjectIdChapterId(subjectId, chapterId);
		
		sqlSession.close();
		return result;
	}
	
	public List<TargetList> getAllTargetBySubjectIdChapterId1(long subjectId, long chapterId){
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		TargetMapper mapper = sqlSession.getMapper(TargetMapper.class);
		QuestionsMapper questionMapper = sqlSession.getMapper(QuestionsMapper.class);
		Questions_TargetMapper questionTargetMapper = sqlSession.getMapper(Questions_TargetMapper.class);
		
		List<Target> targetList = mapper.getBySubjectIdChapterId(subjectId, chapterId);
		List<TargetList> result = new ArrayList<TargetList>();
		
		System.out.println(targetList.size());
		
		for (int i = 0; i < targetList.size(); i++) {
			TargetList target = new TargetList();
			target.target = targetList.get(i);
			target.questionL1Count = 0;
			target.questionL2Count = 0;
			target.questionL3Count = 0;
			target.questionL4Count = 0;
			target.questionL5Count = 0;
			
			List<Questions_TargetKey> questionTargetList = questionTargetMapper.getBySubjectIdChapterIdTargetId(targetList.get(i).getSubjectid(), targetList.get(i).getChapterid(), targetList.get(i).getId());
			List<Questions> questionL1List = questionMapper.getL1BySubjectIdChapterId(targetList.get(i).getSubjectid(), targetList.get(i).getChapterid());
			List<Questions> questionL2List = questionMapper.getL2BySubjectIdChapterId(targetList.get(i).getSubjectid(), targetList.get(i).getChapterid());
			List<Questions> questionL3List = questionMapper.getL3BySubjectIdChapterId(targetList.get(i).getSubjectid(), targetList.get(i).getChapterid());
			List<Questions> questionL4List = questionMapper.getL4BySubjectIdChapterId(targetList.get(i).getSubjectid(), targetList.get(i).getChapterid());
			List<Questions> questionL5List = questionMapper.getL5BySubjectIdChapterId(targetList.get(i).getSubjectid(), targetList.get(i).getChapterid());
		
			for (int j = 0; j < questionL1List.size(); j++) {
				for (int z = 0; z < questionTargetList.size(); z++) {
					if(questionL1List.get(j).getId() == questionTargetList.get(z).getQuestionid()) {
						target.questionL1Count += 1;
					}
				}
			}
			
			for (int j = 0; j < questionL2List.size(); j++) {
				for (int z = 0; z < questionTargetList.size(); z++) {
					if(questionL2List.get(j).getId() == questionTargetList.get(z).getQuestionid()) {
						target.questionL2Count += 1;
					}
				}
			}
			
			for (int j = 0; j < questionL3List.size(); j++) {
				for (int z = 0; z < questionTargetList.size(); z++) {
					if(questionL3List.get(j).getId() == questionTargetList.get(z).getQuestionid()) {
						target.questionL3Count += 1;
					}
				}
			}
			
			for (int j = 0; j < questionL4List.size(); j++) {
				for (int z = 0; z < questionTargetList.size(); z++) {
					if(questionL4List.get(j).getId() == questionTargetList.get(z).getQuestionid()) {
						target.questionL4Count += 1;
					}
				}
			}
			
			for (int j = 0; j < questionL5List.size(); j++) {
				for (int z = 0; z < questionTargetList.size(); z++) {
					if(questionL5List.get(j).getId() == questionTargetList.get(z).getQuestionid()) {
						target.questionL5Count += 1;
					}
				}
			}
			
			result.add(target);
		}
		
		sqlSession.close();
		return result;
	}
	
	@Override
	public List<Chapter> getAllChapterBySubjectId(int subjectId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		ChapterMapper mapper = sqlSession.getMapper(ChapterMapper.class);
		List<Chapter> result = mapper.getAllChapterBySubjectId(subjectId);
		
		sqlSession.close();
		return result;
	}
	
	@Override
	public List<Target> getAllTargetBySubjectId(long subjectId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		TargetMapper mapper = sqlSession.getMapper(TargetMapper.class);
		List<Target> result = mapper.getBySubjectId(subjectId);
		
		sqlSession.close();
		return result;
	}
	
	@Override
	public List<Classes> getAllClassBySubjectId(long teacherId, int subjectId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		ClassesMapper classMapper = sqlSession.getMapper(ClassesMapper.class);
		List<Classes> result = classMapper.getBySubjectId(teacherId, subjectId);
		
		sqlSession.close();	
		return result;
	}
	
	@Override
	public List<Student> getAllStudentBySubjectId(int subjectId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		Student_SubjectMapper studentSubjectMapper = sqlSession.getMapper(Student_SubjectMapper.class);
		List<Student_Subject> studentSubjectList = studentSubjectMapper.getBySubjectId(subjectId);
		
		List<Integer> studentIdList = new ArrayList<Integer>();
		for(int i = 0; i < studentSubjectList.size(); i++) {
			studentIdList.add(studentSubjectList.get(i).getStudentid());
		}
		
		List<Student> result = new ArrayList<Student>();
		if(studentIdList.size() > 0) {
			StudentMapper studentMapper = sqlSession.getMapper(StudentMapper.class);
			result = studentMapper.getByIdInList(studentIdList);
		}

		sqlSession.close();
		return result;
	}
	
	@Override
	public List<Exams> getAllExamsBySubjectId(int subjectId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		ExamsMapper examsMapper = sqlSession.getMapper(ExamsMapper.class);
		List<Exams> result = examsMapper.getBySubjectId(subjectId);
		
		sqlSession.close();
		return result;
	}
	
	@Override
	public int assign(Classes_Exams classExams) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		
		Date now = new Date();
		Classes_ExamsMapper classExamsMapper = sqlSession.getMapper(Classes_ExamsMapper.class);
		classExams.setModifiedat(now);
		classExams.setCreatedat(now);
		int result = classExamsMapper.insertSelective(classExams);
		
		sqlSession.close();
		return result;
	}
	
	@Override
	public int fileStudentToSubject(long subjectId, String content) {
		BufferedReader bufReader = new BufferedReader(new StringReader(content));
		String line = null;
		try {
			while ((line = bufReader.readLine()) != null) {
				System.out.println(line);
			}
		} catch (IOException e) {
			e.printStackTrace();
		}
		return 1;
	}
	
	@Override
	public int removeStudentFromSubject(Student_Subject studentSubject){
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		Student_SubjectMapper mapper = sqlSession.getMapper(Student_SubjectMapper.class);
		Class_MemberMapper classMembetMapper = sqlSession.getMapper(Class_MemberMapper.class);
		
		Class_Member classMember = new Class_Member();
		classMember.setSubjectid(studentSubject.getSubjectid().intValue());
		classMember.setStudentid(studentSubject.getStudentid().longValue());
		classMembetMapper.removeBySubjectId(classMember);
		
		studentSubject.setIsactive(false);
		int result = mapper.updateByPrimaryKeySelective(studentSubject);
		sqlSession.close();
		return result;
	}
	
	@Override
	public int removeStudentFromClass(Class_Member classMember){
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		Class_MemberMapper mapper = sqlSession.getMapper(Class_MemberMapper.class);
		classMember.setIsactive(false);
		int result = mapper.removeByClassId(classMember);
		sqlSession.close();
		return result;
	}
	
	@Override
	public int removeTargetFromSubject(Target target) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		TargetMapper mapper = sqlSession.getMapper(TargetMapper.class);
		target = mapper.getIdTarget(target.getId().intValue());
		target.setIsactive(false);
		int result = mapper.updateByPrimaryKeySelective(target);
		System.out.println("Check Target:" + target.getContent() + target.getIsactive());
		sqlSession.close();
		return result;
	}
	
	@Override
	public int removeChapterFromSubject(Chapter chapter) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		ChapterMapper mapper = sqlSession.getMapper(ChapterMapper.class);
		chapter.setIsactive(false);
		int result = mapper.updateByPrimaryKeySelective(chapter);
		sqlSession.close();
		return result;
	}
	
	@Override
	public int closeClass(Classes classes) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		ClassesMapper mapper = sqlSession.getMapper(ClassesMapper.class);
		classes.setIsopen(false);;
		int result = mapper.updateByPrimaryKeySelective(classes);
		sqlSession.close();
		return result;
	}
	
	@Override
	public List<Student_Exam> fetchExamsResult(long subjectId, long examsId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		Student_ExamMapper mapper = sqlSession.getMapper(Student_ExamMapper.class);
		List<Student_Exam> result = mapper.getBySubjectIdExamsId(subjectId, examsId);
		sqlSession.close();
		return result;
	}
	
	@Override
	public List<Student_Exam> fetchExamsResultClass(long classId, long examsId) {
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		Student_ExamMapper mapper = sqlSession.getMapper(Student_ExamMapper.class);
		List<Student_Exam> result = mapper.getBySubjectIdClassIdExamsId(classId, examsId);
		sqlSession.close();
		return result;
	}

	@Override
	public Subject getSubjectOfExam123(long examsId) {
		// TODO Auto-generated method stub
		SqlSessionFactory sqlMapper = MyBatisService.getSessionFactory();
		SqlSession sqlSession = sqlMapper.openSession(true);
		SubjectMapper subjectMapper  = sqlSession.getMapper(SubjectMapper.class);
		ExamsMapper examsMapper = sqlSession.getMapper(ExamsMapper.class);
	    Exams findExams = new Exams();
	    findExams = examsMapper.selectByPrimaryKey(examsId);
		Subject subject = new Subject();
		
		subject = subjectMapper.selectByPrimaryKey(findExams.getSubjectid().longValue());
		sqlSession.close();
		return subject;
	}
}
